docs/topics/proc_exposing-application-metrics-using-prometheus-with-vertx.adoc

```diff
@@ -95,64 +96,86 @@ import io.micrometer.core.instrument.Meter;
 import io.vertx.core.VertxOptions;
 import io.vertx.core.http.HttpServerOptions;
 import io.vertx.micrometer.MicrometerMetricsOptions;
 import io.vertx.micrometer.VertxPrometheusOptions;
 import io.vertx.micrometer.backends.BackendRegistries;
 
-public class MyVertxApp extends AbstractVerticle {
-
-    @Override
-    public void start() {
-        Vertx vertx = Vertx.vertx(new VertxOptions().setMetricsOptions(
-            new MicrometerMetricsOptions()
-                .setPrometheusOptions(new VertxPrometheusOptions().setEnabled(true)
-                    .setStartEmbeddedServer(true)
-                    .setEmbeddedServerOptions(new HttpServerOptions().setPort(8081))
-                    .setEmbeddedServerEndpoint("/metrics"))
-                .setEnabled(true)));
-
-        PrometheusMeterRegistry registry = (PrometheusMeterRegistry) BackendRegistries.getDefaultNow();
-        registry.config().meterFilter(
-            new MeterFilter() {
-                @Override
-                public DistributionStatisticConfig configure(Meter.Id id, DistributionStatisticConfig config) {
-                return DistributionStatisticConfig.builder()
-                    .percentilesHistogram(true)
-                    .build()
-                    .merge(config);
-                }
-            });
-
-        vertx.deployVerticle(MyVertxApp.class.getName());
-
-        Timer timer = Timer
-            .builder("my.timer")
-            .description("a description of what this timer does")
-            .register(registry);
-
-        vertx.setPeriodic(1000, l -> {
-            timer.record(() -> {
-                vertx.createHttpServer()
-                .requestHandler(req -> req.response().end("Hello"))
-                .listen(8080);
-            });
-        });
-    }
+public class CustomLauncher extends Launcher {
+
+  @Override
+  public void beforeStartingVertx(VertxOptions options) {
+    options.setMetricsOptions(new MicrometerMetricsOptions()
+      .setPrometheusOptions(new VertxPrometheusOptions().setEnabled(true)
+        .setStartEmbeddedServer(true)
+        .setEmbeddedServerOptions(new HttpServerOptions().setPort(8081))
+        .setEmbeddedServerEndpoint("/metrics"))
+      .setEnabled(true));
+  }
+
+  @Override
+  public void afterStartingVertx(Vertx vertx) {
+    PrometheusMeterRegistry registry = (PrometheusMeterRegistry) BackendRegistries.getDefaultNow();
+    registry.config().meterFilter(
+      new MeterFilter() {
+        @Override
+        public DistributionStatisticConfig configure(Meter.Id id, DistributionStatisticConfig config) {
+          return DistributionStatisticConfig.builder()
+            .percentilesHistogram(true)
+            .build()
+            .merge(config);
+        }
+    });
 }
 ----
---
 
-. Set the `<vertx.verticle>` property in the `pom.xml` file of your application to your custom class:
+. Create a custom `AbstractVerticle` class and override the `start` method to collect metrics, for example:
```

Comment:
I would say: _In your application, you can now create your own metrics. For example, here is how to create a `Timer` and use it to measure execution time:_